### PR TITLE
[master] Performance tests improvement - use EmulatedDB

### PR DESCRIPTION
**Pull request by rfelcman**
This change in EclipseLink JPA performance tests replace physical database by emulated one.
Emulation is based on `org.eclipse.persistence.testing.tests.performance.emulateddb.*` classes (JDBC driver).
There are some fixes like new required `EmulatedDatabaseMetaData` class and change in `EmulatedStatement` to distinguish passed SQL queries with parameters.

### CODE REVIEW

File: foundation/org.eclipse.persistence.core.test.framework/src/main/java/org/eclipse/persistence/testing/tests/performance/emulateddb/EmulatedConnection.java [hunk 1]
```diff
@@ -327,10 +328,10 @@ public boolean isClosed() throws SQLException {
      */
     @Override
     public DatabaseMetaData getMetaData() throws SQLException {
-        if (connection != null) {
-            return connection.getMetaData();
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core.test.framework/src/main/java/org/eclipse/persistence/testing/tests/performance/emulateddb/EmulatedConnection.java:331]: if there is an existing connection, metadata should be taken from there; in the other words, this piece should have stayed

File: performance/eclipselink.perf.test/src/test/java/org/eclipse/persistence/testing/perf/jpa/tests/basic/JPAReadAbstract.java [hunk 2]
```diff
@@ -72,25 +92,22 @@ public void testReadEntity() {
...
-            em.getTransaction().commit();
-        } catch (Throwable e) {
-            throw new RuntimeException(e);
-        } finally {
-            if (em.getTransaction().isActive()) {
-                em.getTransaction().rollback();
+    private synchronized void prepareData(EntityManager em) {
+        AbstractSession session = em.unwrap(AbstractSession.class);
+        EmulatedConnection conn = (EmulatedConnection)session.getAccessor().getConnection();
+        Vector<DatabaseField> masterFields = new Vector<>(session.getDescriptor(MasterEntity.class).getAllFields());
```
Comment (lukasj) [inline performance/eclipselink.perf.test/src/test/java/org/eclipse/persistence/testing/perf/jpa/tests/basic/JPAReadAbstract.java:98]: No need to use Vector here

Comment (rfelcman) [reply performance/eclipselink.perf.test/src/test/java/org/eclipse/persistence/testing/perf/jpa/tests/basic/JPAReadAbstract.java:98]: Due a `org.eclipse.persistence.internal.sessions.ArrayRecord#ArrayRecord(java.util.Vector, org.eclipse.persistence.internal.helper.DatabaseField[], java.lang.Object[])` is `java.util.Vector` needed.

File: foundation/org.eclipse.persistence.core.test.framework/src/main/java/org/eclipse/persistence/testing/tests/performance/emulateddb/EmulatedDatabaseMetaData.java [hunk 3]
```diff
@@ -0,0 +1,913 @@
...
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.RowIdLifetime;
+import java.sql.SQLException;
+
+public class EmulatedDatabaseMetaData implements DatabaseMetaData {
+
+    private EmulatedConnection connection;
+
+    public EmulatedDatabaseMetaData(EmulatedConnection emulatedConnection) {
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core.test.framework/src/main/java/org/eclipse/persistence/testing/tests/performance/emulateddb/EmulatedDatabaseMetaData.java:27]: this should probably be a `Connection` instead and where the cast is needed, one should call `unwrap`, WDYT?

Comment (rfelcman) [reply foundation/org.eclipse.persistence.core.test.framework/src/main/java/org/eclipse/persistence/testing/tests/performance/emulateddb/EmulatedDatabaseMetaData.java:27]: I'm not sure if it helps with anything except little bit complicated code. This is our implementation of `DatabaseMetaData` and almost any access to `this.connection` will need cast/unwrap call as specific properties are fetched there.

Comment (lukasj) [reply foundation/org.eclipse.persistence.core.test.framework/src/main/java/org/eclipse/persistence/testing/tests/performance/emulateddb/EmulatedDatabaseMetaData.java:27]: ...and forcing us to use the right API calls in tests... but I'll leave it up to you ;-)

**Pull request merged**